src/main/java/it/gov/pagopa/payhub/activities/activity/paymentsreporting/PaymentsReportingIngestionFlowFileActivityImpl.java

```diff
@@ -91,13 +98,12 @@ private IngestionFlowFileDTO findIngestionFlowFileRecord(Long ingestionFlowFileI
 	 * extracting it from the specified file path.
 	 *
 	 * @param ingestionFlowFileDTO the ingestion flow file DTO containing file details
-	 * @return the extracted {@link File} from the ingestion flow
+	 * @return the extracted {@link List} from the ingestion flow
 	 * @throws IOException if there is an error during file retrieval or extraction
 	 */
-	private File retrieveFile(IngestionFlowFileDTO ingestionFlowFileDTO) throws IOException {
-		List<Path> ingestionFlowFiles = ingestionFlowFileRetrieverService
+	private List<Path> retrieveFile(IngestionFlowFileDTO ingestionFlowFileDTO) throws IOException {
```

Comment:
Why are you changing from a single Path to a List?
If the validation see more than 1 file it would throw an exception (is the validation removing the file in that case?)?